File: compiler-design/module_2/lab_2_3_karanik/src/Position.java
public class Position implements Comparable<Position> {
    private int line;
    private int column;
    private int index;

    public Position(int line, int column, int index) {
        this.line = line;
        this.column = column;
        this.index = index;
    }

    @Override
    public int compareTo(Position other) {
        return Integer.compare(this.index, other.index);
    }

    @Override
    public String toString() {
        return "(" + line + ", " + column + ", " + index + ")";
    }

    public String getSimplifiedString() {
        return "(" + line + ", " + column + ")";
    }
}

File: compiler-design/module_1/lab_1_4_karanik/src/main/kotlin/Position.kt
data class Position(var line: Int, var column: Int, var index: Int) : Comparable<Position> {
    override fun compareTo(other: Position): Int {
        return index.compareTo(other.index)
    }

    override fun toString(): String {
        return "($line, $column, $index)"
    }

    fun getSimplifiedString(): String {
        return "($line, $column)"
    }
}

data class FragmentPosition(var startPosition: Position, var endPosition: Position) {
    override fun toString(): String {
        return "$startPosition-$endPosition"
    }

    fun getSimplifiedString(): String {
        return "${startPosition.getSimplifiedString()}-${endPosition.getSimplifiedString()}"
    }
}